solc/CommandLineInterface.cpp

```diff
@@ -845,52 +845,56 @@ void CommandLineInterface::handleCombinedJSON()
 	output[g_strVersion] = frontend::VersionString;
 	vector<string> contracts = m_compiler->contractNames();
 
+	// NOTE: The state checks here are more strict that in Standard JSON. There we allow
+	// requesting certain outputs even if compilation fails as long as analysis went ok.
+	bool compilationSuccess = m_compiler->state() >= CompilerStack::State::CompilationSuccessful;
+
 	if (!contracts.empty())
 		output[g_strContracts] = Json::Value(Json::objectValue);
 	for (string const& contractName: contracts)
```

Comment:
I can't do it before the loop because after these conditions there's code that does not depend on `compilationSuccess`. All I can do would be to extract the `compilationSuccess` part of this into an outer `if`. That still leaves the individual conditions though and also adds more nesting so not sure if it's really an improvement. I think I'd rather leave it as is.